tests/Unit/IO/Test_VolumeData.py

```diff
@@ -10,159 +10,148 @@
-        expected_xcoord_tensor_data = np.array([
-            0.0, 6.28318530717959, 0.0, 6.28318530717959, 0.0,
-            6.28318530717959, 0.0, 6.28318530717959
-        ])
-        expected_ycoord_tensor_data = np.array([
-            0.0, 0.0, 6.28318530717959, 6.28318530717959, 0.0, 0.0,
-            6.28318530717959, 6.28318530717959
-        ])
-        expected_zcoord_tensor_data = np.array([
-            0.0, 0.0, 0.0, 0.0, 6.28318530717959, 6.28318530717959,
-            6.28318530717959, 6.28318530717959
-        ])
-        # Checking whether two numpy arrays are "almost equal" is easy, so
-        # we convert everything to numpy arrays for comparison.
-        Psi_tensor_data = np.asarray(
-            vol_file.get_tensor_component(observation_id=obs_id,
-                                          tensor_component='Psi'))
-        npt.assert_array_almost_equal(Psi_tensor_data,
-                                      expected_Psi_tensor_data)
-        Error_tensor_data = np.asarray(
-            vol_file.get_tensor_component(observation_id=obs_id,
-                                          tensor_component='Error(Psi)'))
-        npt.assert_array_almost_equal(Error_tensor_data,
-                                      expected_Error_tensor_data)
-        xcoord_tensor_data = np.asarray(
-            vol_file.get_tensor_component(
-                observation_id=obs_id,
-                tensor_component='InertialCoordinates_x'))
-        npt.assert_array_almost_equal(xcoord_tensor_data,
-                                      expected_xcoord_tensor_data)
-        ycoord_tensor_data = np.asarray(
-            vol_file.get_tensor_component(
-                observation_id=obs_id,
-                tensor_component='InertialCoordinates_y'))
-        npt.assert_array_almost_equal(ycoord_tensor_data,
-                                      expected_ycoord_tensor_data)
-        zcoord_tensor_data = np.asarray(
-            vol_file.get_tensor_component(
-                observation_id=obs_id,
-                tensor_component='InertialCoordinates_z'))
-        npt.assert_array_almost_equal(zcoord_tensor_data,
-                                      expected_zcoord_tensor_data)
-        h5_file.close()
+        obs_id = self.vol_file.list_observation_ids()[0]
+        # Test tensor component names
+        tensor_components = set(
+            self.vol_file.list_tensor_components(observation_id=obs_id))
+        expected_tensor_components = ['field_1', 'field_2']
+        self.assertEqual(tensor_components, set(expected_tensor_components))
+        # Test tensor component data
+        for i,expected_tensor_component_and_coord in\
+             enumerate(self.tensor_components_and_coords[2:]):
+            npt.assert_almost_equal(
+                np.asarray(
+                    self.vol_file.get_tensor_component(
+                        observation_id=obs_id,
+                        tensor_component=expected_tensor_components[i]))[0:8],
```

Comment:
If I understand correctly, this tests `field_1` and `field_2` on `grid_1` at the `obs_id`. I thought `obs_id` would be `8435087234` here, so I'm confused why you take the `[2:]` elements of `self.tensor_components_and_coords` above, which should correspond to the second observation id. I think the solution is to 1) please add a brief comment which fields/grid/obsid is being tested and 2) hard-code the `obs_id` in line 130 to the one you want to test, i.e. `8435087234` or `6785087234`.

(BTW, not a change request, but you could just use `0` and `1` instead of the "hashes" as observation IDs)

Reply:
Yeah, list_observation_ids() returns [6785087234, 8435087234]. 
I think I will just change it to 0,1 then.